lib/src/main/java/com/dynatrace/file/util/FilePollerFactory.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2022 Dynatrace LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dynatrace.file.util;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+class FilePollerFactory {
+  private static final Logger logger = Logger.getLogger(FilePollerFactory.class.getName());
+  private static final boolean IS_MAC_OS =
+      System.getProperty("os.name", "").toLowerCase().contains("mac");
+
+  private FilePollerFactory() {}
+
+  /**
+   * See {@link FilePollerFactory#getDefault(String, Duration) getDefault} for details. It calls
+   * this method with the default poll duration of 60 seconds.
+   *
+   * @param fileName The name of the file to be watched. An absolute path will be created from the
+   *     file name if it is not already passed as absolute path.
+   * @return An object that implements the abstract methods on {@link FilePoller}.
+   * @throws IOException if the initialization of the {@link WatchServiceBasedFilePoller} is not
+   *     successful.
+   */
+  static FilePoller getDefault(String fileName) throws IOException {
```

Comment:
Is this ever used? Otherwise the default would be duplicated since you handle do the same if null is passed to the other overload.